### ISSUE TITLE
Side panel: Collapsable heading exploration 3

### ISSUE THREAD

**Issue by bndege2** [carbon-design-system/ibm-products#8952]
This is a follow-on issue to: Side panel: Collapsable heading exploration 2 cont'd https://github.com/carbon-design-system/ibm-products/issues/8893

The new versions of Tearsheet and Page header use a similar collapsable heading to shift between a default and collapsed section. The goal was to keep context as the user scrolls through these two containers and provide access to key actions (such as the close button in Tearsheet).

Side panel has a somewhat similar collapse state with its own animation. However there are key differences, such as the two line wrap when collapsed and the animated transition. We should explore bringing Side panel's collapsed state into alignment with Tearsheet and Page header.

## Tasks

- [ ] Explore what happens when: 
- there’s no title
- there’s no title and a back button?
- when there’s a collapsed heading and any of the above?
- when there’s a title with a back button and a collapsed heading?

- [ ] Identify other permutations that we need to care for
- [ ] Explore behaviour when the panel is collapsed but some other element shows up in the collapsed header instead of the title (the Security example). How do we make this clear to our devs and leave the door open?
- [ ] Explore what slots we need to plan for when designs include tags following the title, or icons in front of the title
- [ ] Explore whether we should enable the lockup of the status icon + title + caption. Should the title be treated as a slot similar to Page header to enable a design like this, or should title be required?
- [ ] Explore whether we need a slot for anything leading the close button, referencing the Instana example where there’s a tag next to the close button
- [ ] Explore the need for a secondary space that collapses other than the title area
- [ ] Explore what could be used in this secondary space (Notification, Tabs, or other useful elements we could recommend)
- [ ] Determine whether the secondary space should adjust its height automatically, given that these elements have different px heights
- [ ] Explore whether elements like the time range example should be in a secondary sticky section or in the title area.
- [ ] Explore how we should handle truncation for the title
- [ ] Explore whether truncation should behave similarly to Tearsheet, where it’s a single line with an ellipsis at the end
- [ ] Determine whether adopters should be able to adjust truncation behaviour, and if so, how spacing should adjust to accommodate
- [ ] Explore whether the collapsing button should be baked in as functionality, noting that not every side panel that collapses should have this
- [ ] Explore how we could guide adopters here (make it themselves in the slot, reference expanding tile as a guide)
- [ ] Explore whether we should enable layouts like the Instana example with no title but a set of tabs and additional buttons, and how we give adopters the space for this section. Do we need to provide accessibility guidance (how is the Side panel announced)?
- [ ] Determine what a “smart default” for collapsing behavior should be when there’s just a title in the header

**Comment by bndege2** [comment]
[Figma file](https://www.figma.com/design/HSAkNzLm3dkBiskozrBdJQ/Side-panel?node-id=35-856&t=Ny3vvnJ2stbYgz9q-1)
Created a [follow-on issue](https://github.com/carbon-design-system/ibm-products/issues/9106) for next phase of explorations
